packages/server/hubspot/hubSpotApi.ts

```diff
@@ -0,0 +1,143 @@
+import fetch from 'node-fetch'
+const accessToken = process.env.HUBSPOT_API_KEY
+
+type Company = {
+  id: string
+  archived: boolean
+}
+
+type DealsResponse = {
+  total: number
+}
+
+type CompaniesResponse = {
+  results: Company[]
+}
+
+const getCompaniesByDomain = async (domain: string): Promise<CompaniesResponse | Error> => {
+  const body = JSON.stringify({
+    properties: ['domain', 'hs_additional_domains'],
+    filterGroups: [
+      {
+        filters: [
+          {
+            operator: 'EQ',
+            propertyName: 'domain',
+            value: domain
+          }
+        ]
+      },
+      {
+        filters: [
+          {
+            operator: 'CONTAINS_TOKEN',
+            propertyName: 'hs_additional_domains',
+            value: domain
+          }
+        ]
+      }
+    ],
+    limit: 100
+  })
+
+  const res = await fetch(`https://api.hubapi.com/crm/v3/objects/companies/search`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${accessToken}`
+    },
+    body
+  })
+
+  const resJson = await res.json()
+
+  if (resJson.status === 'error') {
+    return new Error(resJson.message)
+  }
+
+  return resJson
+}
+
+const SALES_PIPELINE_ID = '14fa7db3-cc82-4562-8603-90d792014b42'
+const PQL_ACCEPTED_STAGE_ID = '2410510'
+const PQL_REPLIED_STAGE_ID = 'd394bd39-f31c-4cac-afba-2d9b85b57e2b'
+const SQL_STAGE_ID = '48c94b2b-1d51-47af-b563-cf490d799ec2'
+const SQL_IN_PROCUREMENT_STAGE_ID = '611924'
```

Comment:
-1 Let's also put these into `.env`. We don't need to distinguish between those, so let's just do
```
HUBSPOT_SALES_PIPELINE_ACTIVE_STAGES='2410510,d394bd39-f31c-4cac-afba-2d9b85b57e2b,48c94b2b-1d51-47af-b563-cf490d799ec2,611924'
```
and then `.split(',')` them before usage.